test/router-luatest/map_part_test.lua

```diff
@@ -22,279 +22,273 @@ local cfg_template = {
+        rawset(_G, 'do_map', function(bids, args, opts)
+            local val, err, err_uuid = ivshard.router.map_part_callrw(
+                bids, 'do_map', args, opts)
+            local val_type
+            if opts.return_raw and val ~= nil then
+                -- Src+value. The src is plain Lua data. The value is raw.
+                local _, one_map = next(val)
+                val_type = type(one_map)
+            else
+                val_type = type(val)
+            end
+            return {
+                val = val,
+                val_type = val_type,
+                err = err,
+                err_uuid = err_uuid,
+            }
+        end)
+        return res, err
     end)
     t.assert(res and not err, 'bootstrap buckets')
+    vtest.cluster_exec_each(cg, function()
+        rawset(_G, 'do_map', function(res)
+            ilt.assert_gt(require('vshard.storage.ref').count, 0)
+            return {ivutil.replicaset_uuid(), res}
+        end)
+        rawset(_G, 'bucket_send', function(bid, dst)
+            local _, err = ivshard.storage.bucket_send(
+                bid, dst, {timeout = iwait_timeout})
+            ilt.assert_equals(err, nil)
+        end)
+    end)
+    cg.rs1_uuid = cg.replica_1_a:replicaset_uuid()
+    cg.rs2_uuid = cg.replica_2_a:replicaset_uuid()
 end)
 
-g.after_all(function()
-    g.cluster:drop()
+g.after_all(function(cg)
+    cg.cluster:drop()
 end)
 
-local function map_part_init()
-    local rs1_uuid = g.replica_1_a:replicaset_uuid()
-    local rs2_uuid = g.replica_2_a:replicaset_uuid()
-
-    local create_map_func_f = function(res1)
-        rawset(_G, 'do_map', function(res2)
-            return {res1, res2}
-        end)
-    end
-    g.replica_1_a:exec(create_map_func_f, {1})
-    g.replica_2_a:exec(create_map_func_f, {2})
-
-    local bids1 = vtest.storage_get_n_buckets(g.replica_1_a, 4)
-    local bids2 = vtest.storage_get_n_buckets(g.replica_2_a, 1)
-
-    return {
-        rs1_uuid = rs1_uuid,
-        rs2_uuid = rs2_uuid,
-        bids1 = bids1,
-        bids2 = bids2,
-    }
+local function router_do_map(router, bids, args, opts)
+    return router:exec(function(bids, args, opts)
+        return _G.do_map(bids, args, opts)
+    end, {bids, args, opts})
 end
 
-g.test_map_part_single_rs = function(g)
-    local expected, res
-    local init = map_part_init()
-
-    res = g.router:exec(function(bid1, bid2)
-        local val, err, err_uuid = ivshard.router.map_part_callrw(
-            {bid1, bid2},
-            'do_map',
-            {3},
-            {timeout = iwait_timeout})
-        return {
-            val = val,
-            err = err,
-            err_uuid = err_uuid,
-        }
-    end, {init.bids1[3], init.bids1[2]})
+g.test_map_part_single_rs = function(cg)
+    local bids = vtest.storage_get_n_buckets(cg.replica_1_a, 4)
+    local res = router_do_map(cg.router, {bids[3], bids[2]},
+                              {123}, {timeout = iwait_timeout})
```

Comment:
Ouch. I think it works in some mode in CI, where unknown globals are `nil`. So those timeouts were all ignored and default. Fixed.

Reply:
Hm, this is super wrong. Thanks for noticing, I've created a ticket. It can't keep happening - https://github.com/tarantool/vshard/issues/492.

Reply:
It still doesn't work. The test is broken on debug version:

```
[001] not ok 6	router.test_map_part_ref_timeout
[001] #   ...mming/tnt/vshard/test/router-luatest/map_callrw_test.lua:319: variable 'test_ref' is not declared
[001] #   stack traceback:
[001] #   	...mming/tnt/vshard/test/router-luatest/map_callrw_test.lua:317: in function 'router.test_map_part_ref_timeout'
[001] #   	...
[001] #   	[C]: in function 'xpcall'
```

Please, let's do smth like that:

```diff
diff --git a/test/luatest_helpers/vtest.lua b/test/luatest_helpers/vtest.lua
index 123a7c8..2418263 100644
--- a/test/luatest_helpers/vtest.lua
+++ b/test/luatest_helpers/vtest.lua
@@ -1,3 +1,5 @@
+require('strict').on()
+
 local t = require('luatest')
 local helpers = require('test.luatest_helpers')
 local cluster = require('test.luatest_helpers.cluster')
```

This way it will fail on release version too

Reply:
Done! I've enabled it in a separate commit.